src/ParallelAlgorithms/LinearSolver/Tags.hpp

```diff
@@ -206,17 +207,21 @@ struct KrylovSubspaceBasis : db::PrefixTag, db::SimpleTag {
  * \brief Holds a `Convergence::HasConverged` flag that signals the linear
  * solver has converged, along with the reason for convergence.
  */
+template <typename OptionsGroup>
 struct HasConverged : db::SimpleTag {
-  static std::string name() noexcept { return "LinearSolverHasConverged"; }
+  static std::string name() noexcept {
+    return option_name<OptionsGroup>() + "(HasConverged)";
```

Comment:
Same goes about format collision with prefix tags (there's also a few more below, but I'll stop commenting on every one and trust that if you change one you'll change them all).